src/contract.rs

```diff
@@ -59,12 +59,17 @@ pub fn instantiate(
     };
 
     let mut total_supply: u128 = 0;
+
+    let prng_seed_hashed = sha_256(&msg.prng_seed.0);
+    PrngStore::save(deps.storage, prng_seed_hashed)?;
+
     {
         let initial_balances = msg.initial_balances.unwrap_or_default();
         for balance in initial_balances {
             let amount = balance.amount.u128();
             let balance_address = deps.api.addr_validate(balance.address.as_str())?;
-            BalancesStore::save(deps.storage, &balance_address, amount)?;
```

Comment:
here save() would do the work, right? saves us `None` stuff which is a bit less readable